lighthouse-core/lib/page-functions.js

```diff
@@ -453,12 +452,13 @@ const getNodeDetailsString = `function getNodeDetails(elem) {
   ${getBoundingClientRect.toString()};
   ${getOuterHTMLSnippet.toString()};
   ${getNodeLabel.toString()};
+  const shadowRootElem = elem instanceof ShadowRoot ? elem.host : elem;
   return {
     devtoolsNodePath: getNodePath(elem),
-    selector: getNodeSelector(elem),
-    boundingRect: getBoundingClientRect(elem),
+    selector: getNodeSelector(shadowRootElem),
+    boundingRect: getBoundingClientRect(shadowRootElem),
     snippet: getOuterHTMLSnippet(elem),
```

Comment:
decided to leave these as is, since they take care of the shadowRoot case. @adamraine @patrickhulce what do y'all think?

Reply:
`getNodePath` will return different results for `ShadowRoot` and `Element`, so leaving that as is was a good call.

@patrickhulce Would you prefer keeping the `ShadowRoot` logic in `getOuterHTMLSnippet` even though it won't be used anywhere after this patch?

Reply:
I agree, I wouldn't go out of our way to remove `ShadowRoot` support from anything just yet, even if the path is currently being covered by our other logic. If we also removed the functions from the public interface of this module, then I'd feel less strongly, but seems like that step would benefit from a more targeted and comprehensive plan for what to do with shadow elements across all of lighthouse that feels out of scope for this PR.